test/e2e/upgrade/upgrade.go

```diff
@@ -157,11 +167,24 @@ var _ = g.Describe("[sig-arch][Feature:ClusterUpgrade]", func() {
 			upgradeTests,
 			func() {
 				for i := 1; i < len(upgCtx.Versions); i++ {
-					framework.ExpectNoError(clusterUpgrade(f, client, dynamicClient, config, upgCtx.Versions[i]), fmt.Sprintf("during upgrade to %s", upgCtx.Versions[i].NodeImage))
+					upgradeErr = clusterUpgrade(f, client, dynamicClient, config, upgCtx.Versions[i])
+					framework.ExpectNoError(upgradeErr, fmt.Sprintf("during upgrade to %s", upgCtx.Versions[i].NodeImage))
 				}
 			},
 		)
 	})
+
+	g.It("Cluster upgrade should complete successfully", func() {
```

Comment:
how is this different than https://github.com/openshift/origin/blob/master/test/e2e/upgrade/upgrade.go#L459-L461?

Reply:
There are cases where all of that stuff doesn't run.  If you fail anywhere before https://github.com/openshift/origin/blob/3a7d25779d8a3c6d21949dddb3c87349f3ef8437/test/e2e/upgrade/upgrade.go#L170, these don't happen.  That's why I was trying to put these test cases in the outer wrapper instead of inside the upgrade runner, and I was hoping [ginkgo.Ordered ](https://onsi.github.io/ginkgo/#ordered-containers) would do that.